src/amoc_rest_utils.erl

```diff
@@ -0,0 +1,173 @@
+-module(amoc_rest_utils).
+
+-export([to_binary/1]).
+-export([to_list/1]).
+-export([to_float/1]).
+-export([to_int/1]).
+-export([to_lower/1]).
+-export([to_upper/1]).
+-export([set_resp_headers/2]).
+-export([to_header/1]).
+-export([to_qs/1]).
+-export([to_binding/1]).
+-export([get_opt/2]).
+-export([get_opt/3]).
+-export([priv_dir/0]).
+-export([priv_dir/1]).
+-export([priv_path/1]).
+
+
+-spec to_binary(iodata() | atom() | number()) -> binary().
+
+to_binary(V) when is_binary(V)  -> V;
+to_binary(V) when is_list(V)    -> iolist_to_binary(V);
+to_binary(V) when is_atom(V)    -> atom_to_binary(V, utf8);
+to_binary(V) when is_integer(V) -> integer_to_binary(V);
+to_binary(V) when is_float(V)   -> float_to_binary(V).
+
+-spec to_list(iodata() | atom() | number()) -> string().
+
+to_list(V) when is_list(V)      -> V;
+to_list(V)                      -> binary_to_list(to_binary(V)).
+
+-spec to_float(iodata()) -> number().
+
+to_float(V) ->
+    Data = iolist_to_binary([V]),
+    case binary:split(Data, <<$.>>) of
+        [Data] ->
+            binary_to_integer(Data);
+        [<<>>, _] ->
+            binary_to_float(<<$0, Data/binary>>);
+        _ ->
+            binary_to_float(Data)
+    end.
+
+%%
+
+-spec to_int(integer() | binary() | list()) -> integer().
+
+to_int(Data) when is_integer(Data) ->
+    Data;
+to_int(Data) when is_binary(Data) ->
+    binary_to_integer(Data);
+to_int(Data) when is_list(Data) ->
+    list_to_integer(Data).
+
+-spec set_resp_headers([{binary(), iodata()}], cowboy_req:req()) -> cowboy_req:req().
+
+set_resp_headers([], Req) ->
+    Req;
+set_resp_headers([{K, V} | T], Req0) ->
+    Req = cowboy_req:set_resp_header(K, V, Req0),
+    set_resp_headers(T, Req).
+
+-spec to_header(iodata() | atom() | number()) -> binary().
+
+to_header(Name) ->
+    Prepared = to_binary(Name),
+    to_lower(Prepared).
+
+-spec to_qs(iodata() | atom() | number()) -> binary().
+
+to_qs(Name) ->
+    to_binary(Name).
+
+-spec to_binding(iodata() | atom() | number()) -> atom().
+
+to_binding(Name) ->
+    Prepared = to_binary(Name),
+    binary_to_atom(Prepared, utf8).
```

Comment:
Do we have control over this? It's dangerous to cast binary to any atom.

Reply:
This is generated code, however AFAIK it's not used anywhere. I tried to keep the changes to the generated code as low as possible